[Bug]: Getting `interface conversion: interface {} is nil, not map[string]interface {}` forcing `restore_to_point_in_time` to `{}` on `aws_db_instance`

### Terraform Core Version

1.5.4

### AWS Provider Version

5.11.0

### Affected Resource(s)

`aws_db_instance`

### Expected Behavior

Providing an empty (`{}`) `restore_to_point_in_time` it should ignore it or treat as missing.

### Actual Behavior

I get this error forcing the variable `restore_to_point_in_time` to `{}`:

`panic: interface conversion: interface {} is nil, not map[string]interface {}`

Checking the code, seems like it tries to cast it to a `map(string)` even if it's `nil`. It should skip it instead or treat it as missing, gracefully.

### Relevant Error/Panic Output Snippet

```shell
panic: interface conversion: interface {} is nil, not map[string]interface {}

github.com/hashicorp/terraform-provider-aws/internal/service/rds/instance.go:1242
```

### Terraform Configuration Files

```terraform
terraform {
  required_version = "= 1.5.4"

  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "= 5.11.0"
    }
  }
}

resource "aws_db_instance" "test_rds" {
  apply_immediately           = true
  identifier                  = "testing-rds"
  allocated_storage           = 10
  engine                      = "postgres"
  engine_version              = "13.11"
  instance_class              = "db.t3.micro"
  manage_master_user_password = true
  username                    = "foo"
  parameter_group_name        = "default.postgres13"

  skip_final_snapshot = true

  restore_to_point_in_time {

  }
}
```

### Steps to Reproduce

`terraform init`
`terraform apply --auto-approve`

### Debug Output

```bash
aws_db_instance.test_rds: Creating...
2023-08-08T18:46:54.653+0200 [INFO]  Starting apply for aws_db_instance.test_rds
2023-08-08T18:46:54.653+0200 [DEBUG] aws_db_instance.test_rds: applying the planned Create change
2023-08-08T18:46:54.655+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: [DEBUG] setting computed for "replicas" from ComputedKeys
2023-08-08T18:46:54.655+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: [DEBUG] setting computed for "tags_all" from ComputedKeys
2023-08-08T18:46:54.655+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: [DEBUG] setting computed for "vpc_security_group_ids" from ComputedKeys
2023-08-08T18:46:54.655+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: [DEBUG] setting computed for "master_user_secret" from ComputedKeys
2023-08-08T18:46:54.655+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: [DEBUG] setting computed for "listener_endpoint" from ComputedKeys
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: panic: interface conversion: interface {} is nil, not map[string]interface {}
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: 
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: goroutine 323 [running]:
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: github.com/hashicorp/terraform-provider-aws/internal/service/rds.resourceInstanceCreate({0x111ce86e0?, 0x14004babfb0}, 0x14004babfb0?, {0x111b54bc0?, 0x140031cd450})
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5:        github.com/hashicorp/terraform-provider-aws/internal/service/rds/instance.go:1242 +0x59c0
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: github.com/hashicorp/terraform-provider-aws/internal/provider.interceptedHandler[...].func1(0x0?, {0x111b54bc0?, 0x140031cd450?})
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5:        github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:111 +0x28c
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x111ce86e0?, {0x111ce86e0?, 0x14002c42f60?}, 0xd?, {0x111b54bc0?, 0x140031cd450?})
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5:        github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:728 +0x64
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14001e94b60, {0x111ce86e0, 0x14002c42f60}, 0x14004576f70, 0x14001666c80, {0x111b54bc0, 0x140031cd450})
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5:        github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:864 +0x8a0
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x1400226f4e8, {0x111ce86e0?, 0x14002c42e40?}, 0x14004c5c140)
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5:        github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/grpc_provider.go:1024 +0xb78
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0x140031b7e30, 0x1400169e670, 0x140031b7e60, 0x140031b7e90, {0x140031a33c0, 0x2, 0x2}}, {0x111ce86e0?, 0x14002c42a50?}, 0x14004c5c140)
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5:        github.com/hashicorp/terraform-plugin-mux@v0.11.2/tf5muxserver/mux_server_ApplyResourceChange.go:30 +0x100
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x140031c1040, {0x111ce86e0?, 0x14002c42210?}, 0x1400023e150)
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5:        github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/tf5server/server.go:821 +0x3bc
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1118de2c0?, 0x140031c1040}, {0x111ce86e0, 0x14002c42210}, 0x1400023e070, 0x0)
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5:        github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:422 +0x170
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: google.golang.org/grpc.(*Server).processUnaryRPC(0x14000060780, {0x111cf53e0, 0x14002c25ba0}, 0x14002c36120, 0x14000177d10, 0x117c2d040, 0x0)
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5:        google.golang.org/grpc@v1.56.1/server.go:1337 +0xc90
2023-08-08T18:46:54.658+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: google.golang.org/grpc.(*Server).handleStream(0x14000060780, {0x111cf53e0, 0x14002c25ba0}, 0x14002c36120, 0x0)
2023-08-08T18:46:54.659+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5:        google.golang.org/grpc@v1.56.1/server.go:1714 +0x82c
2023-08-08T18:46:54.659+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: google.golang.org/grpc.(*Server).serveStreams.func1.1()
2023-08-08T18:46:54.659+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5:        google.golang.org/grpc@v1.56.1/server.go:959 +0x84
2023-08-08T18:46:54.659+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5: created by google.golang.org/grpc.(*Server).serveStreams.func1
2023-08-08T18:46:54.659+0200 [DEBUG] provider.terraform-provider-aws_v5.11.0_x5:        google.golang.org/grpc@v1.56.1/server.go:957 +0x16c
2023-08-08T18:46:54.667+0200 [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-08-08T18:46:54.667+0200 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.11.0/darwin_arm64/terraform-provider-aws_v5.11.0_x5 pid=38245 error="exit status 2"
2023-08-08T18:46:54.667+0200 [ERROR] plugin.(*GRPCProvider).ApplyResourceChange: error="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-08-08T18:46:54.667+0200 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-08-08T18:46:54.673+0200 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2023-08-08T18:46:54.673+0200 [ERROR] vertex "aws_db_instance.test_rds" error: Plugin did not respond
2023-08-08T18:46:54.677+0200 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2023-08-08T18:46:54.677+0200 [ERROR] vertex "module.rds_postgresql.module.db_parameter_group.aws_db_parameter_group.this[0] (destroy)" error: Plugin did not respond
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│ 
│   with aws_db_instance.test_rds,
│   on main.tf line 12, in resource "aws_db_instance" "test_rds":
│   12: resource "aws_db_instance" "test_rds" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-aws_v5.11.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 323 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/rds.resourceInstanceCreate({0x111ce86e0?, 0x14004babfb0}, 0x14004babfb0?, {0x111b54bc0?, 0x140031cd450})
        github.com/hashicorp/terraform-provider-aws/internal/service/rds/instance.go:1242 +0x59c0
github.com/hashicorp/terraform-provider-aws/internal/provider.interceptedHandler[...].func1(0x0?, {0x111b54bc0?, 0x140031cd450?})
        github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:111 +0x28c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x111ce86e0?, {0x111ce86e0?, 0x14002c42f60?}, 0xd?, {0x111b54bc0?, 0x140031cd450?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:728 +0x64
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14001e94b60, {0x111ce86e0, 0x14002c42f60}, 0x14004576f70, 0x14001666c80, {0x111b54bc0, 0x140031cd450})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:864 +0x8a0
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x1400226f4e8, {0x111ce86e0?, 0x14002c42e40?}, 0x14004c5c140)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/grpc_provider.go:1024 +0xb78
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0x140031b7e30, 0x1400169e670, 0x140031b7e60, 0x140031b7e90, {0x140031a33c0, 0x2, 0x2}}, {0x111ce86e0?, 0x14002c42a50?}, 0x14004c5c140)
        github.com/hashicorp/terraform-plugin-mux@v0.11.2/tf5muxserver/mux_server_ApplyResourceChange.go:30 +0x100
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x140031c1040, {0x111ce86e0?, 0x14002c42210?}, 0x1400023e150)
        github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/tf5server/server.go:821 +0x3bc
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1118de2c0?, 0x140031c1040}, {0x111ce86e0, 0x14002c42210}, 0x1400023e070, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:422 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000060780, {0x111cf53e0, 0x14002c25ba0}, 0x14002c36120, 0x14000177d10, 0x117c2d040, 0x0)
        google.golang.org/grpc@v1.56.1/server.go:1337 +0xc90
google.golang.org/grpc.(*Server).handleStream(0x14000060780, {0x111cf53e0, 0x14002c25ba0}, 0x14002c36120, 0x0)
        google.golang.org/grpc@v1.56.1/server.go:1714 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        google.golang.org/grpc@v1.56.1/server.go:959 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.56.1/server.go:957 +0x16c

Error: The terraform-provider-aws_v5.11.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

2023-08-08T18:46:54.683+0200 [DEBUG] provider: plugin exited
```

### Panic Output

```
Stack trace from the terraform-provider-aws_v5.11.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 323 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/rds.resourceInstanceCreate({0x111ce86e0?, 0x14004babfb0}, 0x14004babfb0?, {0x111b54bc0?, 0x140031cd450})
        github.com/hashicorp/terraform-provider-aws/internal/service/rds/instance.go:1242 +0x59c0
github.com/hashicorp/terraform-provider-aws/internal/provider.interceptedHandler[...].func1(0x0?, {0x111b54bc0?, 0x140031cd450?})
        github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:111 +0x28c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x111ce86e0?, {0x111ce86e0?, 0x14002c42f60?}, 0xd?, {0x111b54bc0?, 0x140031cd450?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:728 +0x64
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x14001e94b60, {0x111ce86e0, 0x14002c42f60}, 0x14004576f70, 0x14001666c80, {0x111b54bc0, 0x140031cd450})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/resource.go:864 +0x8a0
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x1400226f4e8, {0x111ce86e0?, 0x14002c42e40?}, 0x14004c5c140)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.27.0/helper/schema/grpc_provider.go:1024 +0xb78
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0x140031b7e30, 0x1400169e670, 0x140031b7e60, 0x140031b7e90, {0x140031a33c0, 0x2, 0x2}}, {0x111ce86e0?, 0x14002c42a50?}, 0x14004c5c140)
        github.com/hashicorp/terraform-plugin-mux@v0.11.2/tf5muxserver/mux_server_ApplyResourceChange.go:30 +0x100
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x140031c1040, {0x111ce86e0?, 0x14002c42210?}, 0x1400023e150)
        github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/tf5server/server.go:821 +0x3bc
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1118de2c0?, 0x140031c1040}, {0x111ce86e0, 0x14002c42210}, 0x1400023e070, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:422 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0x14000060780, {0x111cf53e0, 0x14002c25ba0}, 0x14002c36120, 0x14000177d10, 0x117c2d040, 0x0)
        google.golang.org/grpc@v1.56.1/server.go:1337 +0xc90
google.golang.org/grpc.(*Server).handleStream(0x14000060780, {0x111cf53e0, 0x14002c25ba0}, 0x14002c36120, 0x0)
        google.golang.org/grpc@v1.56.1/server.go:1714 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        google.golang.org/grpc@v1.56.1/server.go:959 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.56.1/server.go:957 +0x16c

Error: The terraform-provider-aws_v5.11.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

2023-08-08T18:46:54.683+0200 [DEBUG] provider: plugin exited
```

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

Yes

--- a/.changelog/32928.txt
+++ b/.changelog/32928.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_db_instance: Fix crash creating resource with empty `restore_to_point_in_time` configuration block
+```

--- a/internal/service/rds/cluster.go
+++ b/internal/service/rds/cluster.go
@@ -798,7 +798,8 @@ func resourceClusterCreate(ctx context.Context, d *schema.ResourceData, meta int
 		if err != nil {
 			return sdkdiag.AppendErrorf(diags, "creating RDS Cluster (restore from S3) (%s): %s", identifier, err)
 		}
-	} else if v, ok := d.GetOk("restore_to_point_in_time"); ok {
+	} else if v, ok := d.GetOk("restore_to_point_in_time"); ok && v.([]interface{})[0] != nil {
+
 		tfMap := v.([]interface{})[0].(map[string]interface{})
 		input := &rds.RestoreDBClusterToPointInTimeInput{
 			DBClusterIdentifier:       aws.String(identifier),

--- a/internal/service/rds/cluster.go
+++ b/internal/service/rds/cluster.go
@@ -799,7 +799,6 @@ func resourceClusterCreate(ctx context.Context, d *schema.ResourceData, meta int
 			return sdkdiag.AppendErrorf(diags, "creating RDS Cluster (restore from S3) (%s): %s", identifier, err)
 		}
 	} else if v, ok := d.GetOk("restore_to_point_in_time"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
-
 		tfMap := v.([]interface{})[0].(map[string]interface{})
 		input := &rds.RestoreDBClusterToPointInTimeInput{
 			DBClusterIdentifier:       aws.String(identifier),

--- a/internal/service/rds/cluster.go
+++ b/internal/service/rds/cluster.go
@@ -798,7 +798,7 @@ func resourceClusterCreate(ctx context.Context, d *schema.ResourceData, meta int
 		if err != nil {
 			return sdkdiag.AppendErrorf(diags, "creating RDS Cluster (restore from S3) (%s): %s", identifier, err)
 		}
-	} else if v, ok := d.GetOk("restore_to_point_in_time"); ok && v.([]interface{})[0] != nil {
+	} else if v, ok := d.GetOk("restore_to_point_in_time"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
 
 		tfMap := v.([]interface{})[0].(map[string]interface{})
 		input := &rds.RestoreDBClusterToPointInTimeInput{

--- a/internal/service/rds/instance.go
+++ b/internal/service/rds/instance.go
@@ -1238,7 +1238,7 @@ func resourceInstanceCreate(ctx context.Context, d *schema.ResourceData, meta in
 			output := outputRaw.(*rds.RestoreDBInstanceFromDBSnapshotOutput)
 			resourceID = aws.StringValue(output.DBInstance.DbiResourceId)
 		}
-	} else if v, ok := d.GetOk("restore_to_point_in_time"); ok && v.([]interface{})[0] != nil {
+	} else if v, ok := d.GetOk("restore_to_point_in_time"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
 
 		tfMap := v.([]interface{})[0].(map[string]interface{})
 

--- a/internal/service/rds/instance.go
+++ b/internal/service/rds/instance.go
@@ -1239,9 +1239,7 @@ func resourceInstanceCreate(ctx context.Context, d *schema.ResourceData, meta in
 			resourceID = aws.StringValue(output.DBInstance.DbiResourceId)
 		}
 	} else if v, ok := d.GetOk("restore_to_point_in_time"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
-
 		tfMap := v.([]interface{})[0].(map[string]interface{})
-
 		input := &rds.RestoreDBInstanceToPointInTimeInput{
 			AutoMinorVersionUpgrade:    aws.Bool(d.Get("auto_minor_version_upgrade").(bool)),
 			CopyTagsToSnapshot:         aws.Bool(d.Get("copy_tags_to_snapshot").(bool)),

--- a/internal/service/rds/instance.go
+++ b/internal/service/rds/instance.go
@@ -1238,7 +1238,8 @@ func resourceInstanceCreate(ctx context.Context, d *schema.ResourceData, meta in
 			output := outputRaw.(*rds.RestoreDBInstanceFromDBSnapshotOutput)
 			resourceID = aws.StringValue(output.DBInstance.DbiResourceId)
 		}
-	} else if v, ok := d.GetOk("restore_to_point_in_time"); ok {
+	} else if v, ok := d.GetOk("restore_to_point_in_time"); ok && v.([]interface{})[0] != nil {
+
 		tfMap := v.([]interface{})[0].(map[string]interface{})
 
 		input := &rds.RestoreDBInstanceToPointInTimeInput{

